src/main/java/com/checkmarx/flow/service/GitAuthUrlGenerator.java

```diff
@@ -0,0 +1,30 @@
+package com.checkmarx.flow.service;
+
+import com.checkmarx.flow.dto.ScanRequest;
+import com.checkmarx.flow.exception.MachinaRuntimeException;
+import com.checkmarx.sdk.config.Constants;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class GitAuthUrlGenerator {
+
+    public String overrideGitAuthUrlByScmAccessToken(ScanRequest.Repository repoType, String gitUrl, String scmAccessToken) {
```

Comment:
Maybe a better name would be `addCredentialsToUrl`?

Reply:
Done.